src/test/java/com/epam/test_generator/dao/SuitDAOTest.java

```diff
@@ -141,26 +180,50 @@ public void testRemoveList() {
     }
 
     private Suit retriveSuit() {
-        SimpleDateFormat formatter = new SimpleDateFormat("MM/dd/yyyy");
+        final SimpleDateFormat formatter = new SimpleDateFormat("MM/dd/yyyy");
         return new Suit("Suit1", "Suit1 description", 3,
-                formatter.format(Calendar.getInstance().getTime()), "tag1,tag2", new ArrayList<>());
+                formatter.format(Calendar.getInstance().getTime()), retrieveTagList("tag1","tag2"), new ArrayList<>());
     }
 
     private List<Suit> retrieveSuiteList() {
-        SimpleDateFormat formatter = new SimpleDateFormat("MM/dd/yyyy");
-        Suit suit1 = new Suit("Suit1", "Suit1 description", 5,
-                formatter.format(Calendar.getInstance().getTime()), "tag1 tag2", new ArrayList<>());
+        final SimpleDateFormat formatter = new SimpleDateFormat("MM/dd/yyyy");
+        final Suit suit1 = new Suit("Suit1", "Suit1 description", 5,
+                formatter.format(Calendar.getInstance().getTime()), retrieveTagList("tag1","tag2"), new ArrayList<>());
 
-        Suit suit2 = new Suit("Suit2", "Suit2 description", 5,
-                formatter.format(Calendar.getInstance().getTime()), "tag1 ", new ArrayList<>());
+        final Suit suit2 = new Suit("Suit2", "Suit2 description", 5,
+                formatter.format(Calendar.getInstance().getTime()), retrieveTagList("tag1"), new ArrayList<>());
 
-        Suit suit3 = new Suit("Suit3", "Suit3 description", 5,
-                formatter.format(Calendar.getInstance().getTime()), "tag1 tag3", new ArrayList<>());
+        final Suit suit3 = new Suit("Suit3", "Suit3 description", 5,
+                formatter.format(Calendar.getInstance().getTime()), retrieveTagList("tag1","tag3"), new ArrayList<>());
 
-        ArrayList<Suit> suits = new ArrayList<>();
+        final List<Suit> suits = new ArrayList<>();
         suits.add(suit1);
         suits.add(suit2);
         suits.add(suit3);
         return suits;
     }
+
+    private Set<Tag> retrieveTagList(String ... tags){
+        return Stream.of(tags).map(Tag::new).collect(Collectors.toSet());
+    }
+
+    private Set<Tag> setIdsForTags(Set<Tag> tagsWithIds, Set<Tag> tagsWithoutIds) {
```

Comment:
If I understand correctly, then retrieveSuiteList() returns Suit instances which don't have any ids, because  ##.save()  is responsible for their generation. But setIdsForTags method is in charge of extraction of real ids from each saved Tag from each saved Suit and also it's responsible for filling these ids into tags. So it means that these methods have deferent logic. And I don't understand a reason for moving setIdsForTags into retrieveSuitList().

Correct me if I'm wrong.

Reply:
Ok got it.